testsuite/forge/src/backend/k8s_deployer/deployer.rs

```diff
@@ -0,0 +1,444 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    FORGE_DEPLOYER_IMAGE_TAG, FORGE_DEPLOYER_SERVICE_ACCOUNT_NAME,
+    FORGE_DEPLOYER_VALUES_ENV_VAR_NAME, FORGE_INDEXER_DEPLOYER_DOCKER_IMAGE_REPO,
+    FORGE_TESTNET_DEPLOYER_DOCKER_IMAGE_REPO,
+};
+use crate::{maybe_create_k8s_resource, K8sApi, ReadWrite, Result};
+use k8s_openapi::api::{
+    batch::v1::Job,
+    core::v1::{ConfigMap, Namespace, ServiceAccount},
+    rbac::v1::RoleBinding,
+};
+use kube::{
+    api::{ObjectMeta, PostParams},
+    ResourceExt,
+};
+use serde::{Deserialize, Serialize};
+use std::{collections::BTreeMap, fmt, sync::Arc};
+
+/// These are the values that the forge deployer needs to deploy the forge components to the k8s cluster.
+/// There are global values such as profile, era, and namespace as well as component-specific values
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct ForgeDeployerValues {
+    pub profile: String,
+    pub era: String,
+    pub namespace: String,
+
+    // component specific values
+    // TODO: add an options reference. Ideally this customization is almost always optional and instead handled by the profiles
+    pub indexer_grpc_values: Option<serde_json::Value>,
+    pub indexer_processor_values: Option<serde_json::Value>,
+}
+
+/// The ForgeDeployerManager is responsible for managing the lifecycle of forge deployers, wihch deploy the
+/// forge components to the k8s cluster.
+pub struct ForgeDeployerManager {
```

Comment:
A couple of thoughts on ForgeDeployerManager 

1) ForgeDeployerManager doesn't actually need a type, it would make more sense to have that built in to the ForgeDeployerValues and either have an enum or a grpc that could handle the serialization / deserialization or just have a less strongly typed map of values. Consider either
```
struct ForgeBackendConfig {
  namespace: String, // Namespace to run in? Not sure if this even should be a property you specify, should forge decide this for you?
  image: String, // Full qualifier of the backend image
  values:  serde_json::Value,
}
```

Because in internal-ops you're deserializing this, you either want to have no strict serialization (since you already just render this to json) or you should use grpc. Right now you kind of have the restriction of types without the actual enforcement and the downside of backwards compat issues if you change the struct on either side (grpc would allow to handle that more safely)

2) This is probably more accurately called "ForgeBackend" instead of DeployerManager, since effectively you're just instantiating an arbitrary backend, I would avoid any specificity around the indexer vs other types since it doesnt actually seem like you're using the types here? Just deciding what image to use which could be passed instead by the config.